src/main/java/uk/gov/justice/digital/config/JobArguments.java

```diff
@@ -0,0 +1,162 @@
+package uk.gov.justice.digital.config;
+
+import jakarta.inject.Inject;
+import jakarta.inject.Singleton;
+import io.micronaut.context.ApplicationContext;
+import io.micronaut.context.env.CommandLinePropertySource;
+import io.micronaut.context.env.MapPropertySource;
+import io.micronaut.context.env.PropertySource;
+import lombok.val;
+import org.apache.spark.streaming.Duration;
+import org.apache.spark.streaming.Durations;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.util.AbstractMap.SimpleEntry;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+/**
+ * Class that defines and provides access to the job arguments that we support.
+ *
+ * Arguments are taken from the values parsed by the CommandLinePropertySource.
+ */
+@Singleton
+public class JobArguments {
+
+    private static final Logger logger = LoggerFactory.getLogger(JobArguments.class);
+
+    public static final String AWS_DYNAMODB_ENDPOINT_URL = "dpr.aws.dynamodb.endpointUrl";
```

Comment:
can we move all static final to an interface and implement the interface in JobArguments?

Reply:
Unfortunately not since static final members cannot be declared on Java interfaces.